### ISSUE TITLE
Local-users-only forums do not support interaction (like/comment) on the forum's status panel

### ISSUE THREAD

**Issue by dfsklar** [friendica/friendica#3835]
The concept of a "forum" is -- if I understand correctly -- is intended to allow there to be a single page that members of the forum can visit to see all activity related to the forum, not just their own activity or the forum leader's activity, but *all* posts that were made "via forum XYZ".

I have proof that this is not occurring.  There seems to be no way to visit a single page and see all the forum's activities by all members, not just the leader.

IMAGE 1:  From the forum leader's POV, we see a post that he just made ... and we also see a post made "via" the forum by a regular member of the forum ("mutual friend", ref level of 3).

![screen shot 2017-10-29 at 5 11 52 pm](https://user-images.githubusercontent.com/1151882/32148380-6a3ce8da-bccc-11e7-96da-4e84692f4c79.png)

IMAGE 2:  From that same regular member's POV, we now see the forum's "network" page having the special URL `.../network?f=&cid=19`.  We see the edit box which makes it clear that the forum is well aware that this user is a read-write member of the forum.  And we see the post the forum leader just made.  But we are unable to see the post *this* user made via this same forum (via this same webpage) earlier today.  

![screen shot 2017-10-29 at 5 17 30 pm](https://user-images.githubusercontent.com/1151882/32148426-2372205e-bccd-11e7-9cd2-859e35a8a033.png)

I.e. the "network page" for a FORUM is not really showing all the posts made by the forum leader *and* the posts made "via" that forum (by all forum members).

**Comment by dfsklar** [comment]
There are some folks who might say that the forum's *status* page, not its *network* page, is the place to go to see all postings related to the forum. 

And indeed as the below screenshot shows, we do end up seeing all the posts when we visit the forum's status page.

But this is unacceptable because that page does not support interaction.  As you can see from the screenshot, I am not being invited to like/comment... and I am not being allowed to initiate a new post either.  

The status page needs to be made interactive, or the network page needs to be made complete.  

![screen shot 2017-10-29 at 5 20 39 pm](https://user-images.githubusercontent.com/1151882/32148482-02415b56-bcce-11e7-81d1-d385367c1cbc.png)

**Comment by dfsklar** [comment]
I'll be happy to do the coding to repair... but I need guidance from veterans of this codebase re: the best way to solve, particularly since this may be due to a decision decision that needs re-thinking...

**Comment by tobiasd** [comment]
> But this is unacceptable because that page does not support
> interaction.  As you can see from the screenshot, I am not being
> invited to like/comment... and I am not being allowed to initiate a
> new post either.  

It does support interaction if you follow a "magic link" from your node to the forums profile. The mouse icon will change, when your find such a magic link. Following it you will be authentificated on the other node and can interact with the postings there.

**Comment by annando** [comment]
The whole forum thing is a little bit more complicated. You can do a public post and mention a group in the post. Then the post is distributed to all of your followers and **additionally** to the followers of the forum. Or you only post to the forum. Then only the forum followers will receive the post.

For you, it will always be a post that was made by you. For third parties it depends upon if they are followers of you or of the forum. And if they are following both you and the forum then it depends which posts arrives first.

I want to improve this. But currently I'm unsure, how to do it. This could possibly be done with a change in the item storage.

**Comment by dfsklar** [comment]
@tobiasd made a very good point:  I should try to interact with the "status" panel of the official support forum located at https://forum.friendi.ca/profile/helpers

I did so indeed.  My account is considered a "fan"-level contact of that support forum.

I find that I can *LIKE* anything (post or comment) on the forum.  But I cannot *comment*.  (Why is that?  Comment interaction is important - "like" is not always sufficient.)

NOW: what is really bothering me is that on my own 3.6-dev installation of friendica server, my forum account's status panel does NOT allow any reactions of any kind (not likes, not comments) when a friend visits it, even if the friend is of "mutual friendship" level (not merely "fan" level).  

Is the official friendica support forum using a bleeding-edge version?  Or an old but stable one?  How can I determine how to "match" the way the support forum was created/configured?  I simply want my own server's public forums to show the same UX to visitors, i.e. like/comment support on all posts appearing on the "status" panel obtained via the URL `.../profile/{forumname}`.

**Comment by AlfredSK** [comment]
Let's try to make this a little easier or less confusing (I hope so). :)
In my Friendica workflow I never ever go to other nodes or remote profile pages. I do all reading, writing, commenting and liking while I'm on my own node/profile. The same is true for forums (which are in fact normal accounts with some special functionality).
If I am a fan of a forum I just click on the forum's name in the left pane widget on my network page. If I do so I see all posts and comments I received from the forum and I can interact with them.

**Comment by rabuzarus** [comment]
Something about Magic Links:
When a connection (e.g. friendship) between two friendica contacts is established, two secret keys were created. These keys were used to authenticate the contact.
Now the normal link to a established contact will be replaced by a  internal link (e.g. `..../redir/24` - where 24 is the internal contact ID). So you will redirected to `redir`. `redir` does look into your database if you are really connected to the requested contact. If the db request was successful, it redirects to the requested contact profile-page. In addition it does submit the authentication key to the requested profile. The requested profile instance does check if the key is correct. If the key is correct you should be able two write (if the owner has given permission to write in the settings).

To summarize: all links on your friendica instance which look like `..../redir/24` (the number can be different) are magic links

**Comment by dfsklar** [comment]
I have made progress on diagnostics.  The problem is that my forum is a LOCAL-only -- all members of my forum are accounts *on* the same FR host as the forum itself.  My userbase will all be "local", not remote.  This exposes a bug(?) in function `function can_write_wall`:  it always returns false for a local user.  It cannot possibly be the case that local users are intended to be "discriminated against" in this way 😄 !  I'm quite sure this is a logic error.  I can't see how remoteness is necessary to allow forum members to perform reactive actions directly on the forum's status page.

**Comment by annando** [comment]
The function ```can_write_wall``` _should_ work with a user on the same server as well. This user would be a "remote user" as well.

**Comment by dfsklar** [comment]
Well, I can vouch for the fact that `remote` was false and the function failed.  After I repaired that bug with a simple test hardwiring of `can_write_wall` returning true unconditionally, the bug that I'm describing in this issue report disappeared!  I'm a happy camper and able to proceed with development... but sad that I must make such a bizarre change to the code without explanation.

**Comment by dfsklar** [comment]
sorry did not mean to close the issue.  I'm hoping that this will be considered a real defect and repaired by the dev community.  I am at your disposal and will help in any way possible to determine why `remote` is false for my local users.

**Comment by annando** [comment]
You created that much issues that it will take some time to work on them :-)

**Comment by dfsklar** [comment]
👍 I hope that's a good thing, Annando!  I'm happy to be a member of the community, and will be happy to construct PRs but of course the "history" and "culture" of this codebase is unknown to me so I will need mentorship.

**Comment by annando** [comment]
In some weeks we will have our yearly hackathon in Berlin. We will have something to do, I guess.

**Comment by dfsklar** [comment]
Along with my repair to related issue #3843 (PR coming soon), I now have a local friendica server that allows all members of the forum to successfully post comments directly on the forum's status wall and see each other's comments, like actions, etc.  So there is hope; it is possible to create a forum wall that shows everything from all members and allows all the necessary types of reaction activities.

**Comment by tobiasd** [comment]
> The problem is that my forum is a LOCAL-only -- all members of my
> forum are accounts *on* the same FR host as the forum itself.

I think the problem is, what does local and remote mean. IIRC everything that is not you is considered remote independently on which node in the network it is located at.

**Comment by annando** [comment]
I did several PR to the develop branch that should make the forum handling easier. At first the most important thing is what @AlfredSK already said: Mostly you don't use the forum on the forum's page but in your regular account.

Creating a forum post is like creating a regular post - with a single exception: You mention the forum with a ```@``` or (this is new) with a ```!```. So for example you would do a forum post to the forum ```forum@hostname.tld``` by simply adding  ```!forum@hostname.tld``` to your regular post. Then this post is distributed to all people who subscribed to this forum. This post will appear on the ```/network``` pages of each of these users and they can interact like with any other post. See a Friendica forum as some kind of relay for posts and not so much like a bulletin board forum.

It is recommended that users enable the list of forums (in the settings under ```/settings/features``` it is the option that is labelled "Enable widget to display the forums your are connected with"). Now the users can see the forums they are connected with on the left side. By clicking on the name, they see a list of the postings of this forum. By clicking on the icon (instead of the name) users jump to the remote forum profile page. It is called "remote" since it isn't your profile page. It doesn't matter if it is on your server or another one.

Interaction on the remote forum profile page should be possible as well but depends on the settings of the forum.

Concerning the relation between user and forum: It is correct that the connection is unidirectional. The users want to get all content from the forum they subscribed to. But of course the forum doesn't want any post from the users, but only the content that should be relayed by the forum.
